### PR TITLE
[conditional-exception] conditional throwing

### PR DESCRIPTION
**Pull request by sk1418**

### CODE REVIEW

File: core-kotlin-modules/core-kotlin-exceptions/src/test/kotlin/com/baeldung/conditionalThrowing/ConditionalThrowingUnitTest.kt [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+class InvalidPlayerException(message: String) : RuntimeException(message)
+
+class ConditionalThrowingUnitTest {
+
+    @Test
+    fun `when using require() then get expected results`() {
+        val str = "a b c"
+        assertThrows<IllegalArgumentException> {
+            require(str.length > 10) { "The string is too short." }
+            null.toString()
```
Comment (theangrydev) [inline core-kotlin-modules/core-kotlin-exceptions/src/test/kotlin/com/baeldung/conditionalThrowing/ConditionalThrowingUnitTest.kt:32]: Can we remove this null.toString()?

File: core-kotlin-modules/core-kotlin-exceptions/src/test/kotlin/com/baeldung/conditionalThrowing/ConditionalThrowingUnitTest.kt [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+    @Test
+    fun `when using takeIf() then get expected results`() {
+        val str = "a b c"
+        assertThrows<MyException> {
+            str.takeIf { it.length > 10 } ?: throw MyException("The string is too short.")
+        }
+
+        val nullIsValid: String? = null
+        // we don't expect the exception
```
Comment (theangrydev) [inline core-kotlin-modules/core-kotlin-exceptions/src/test/kotlin/com/baeldung/conditionalThrowing/ConditionalThrowingUnitTest.kt:77]: Let's remove this comment too, it's a bit confusing to say in a comment we don't expect the exception then have an assertion showing there is an exception. Or reword the comment to clarify what you mean.

**Review by theangrydev — CHANGES_REQUESTED** [review]

File: core-kotlin-modules/core-kotlin-exceptions/src/test/kotlin/com/baeldung/conditionalThrowing/ConditionalThrowingUnitTest.kt (hunk 1 above)
Comment (sk1418) [reply core-kotlin-modules/core-kotlin-exceptions/src/test/kotlin/com/baeldung/conditionalThrowing/ConditionalThrowingUnitTest.kt]: ✅ removed

File: core-kotlin-modules/core-kotlin-exceptions/src/test/kotlin/com/baeldung/conditionalThrowing/ConditionalThrowingUnitTest.kt (hunk 2 above)
Comment (sk1418) [reply core-kotlin-modules/core-kotlin-exceptions/src/test/kotlin/com/baeldung/conditionalThrowing/ConditionalThrowingUnitTest.kt]: ✅ removed

**Review by theangrydev — APPROVED** [review]

**Pull request merged**